src/Symfony/Component/Messenger/Command/FailedMessagesRemoveCommand.php

```diff
@@ -69,6 +70,24 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $shouldDeleteAllMessages = $input->getOption('all');
 
         $idsCount = \count($ids);
+
+        if (!$receiver instanceof ListableReceiverInterface) {
+            throw new RuntimeException(\sprintf('The "%s" receiver does not support removing specific messages.', $failureTransportName));
+        }
+
+        if (!$idsCount && null !== $input->getOption('class-filter')) {
+            $ids = $this->getMessageIdsByClassFilter($input->getOption('class-filter'), $receiver);
+            $idsCount = \count($ids);
+
+            if (!$idsCount) {
+                throw new RuntimeException('No failed messages were found with this filter.');
+            }
+
+            if (!$io->confirm(\sprintf('There is %d message%s to remove. Do you want to continue?', $idsCount, $idsCount > 1 ? 's' : ''))) {
```

Comment:
To avoid using `is` or `are` depending on the number of messages, let's rewrite it to something like:

```suggestion
            if (!$io->confirm(\sprintf('Can you confirm you want to remove %d message%s?', $idsCount, 1 === $idsCount ? '' : 's'))) {
```